hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/common/LinkFactoryImpl.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.restjava.common;
+
+import static com.hedera.mirror.restjava.common.EntityIdNumParameter.ENTITY_ID_PATTERN;
+import static com.hedera.mirror.restjava.common.ParameterNames.LIMIT;
+import static com.hedera.mirror.restjava.common.ParameterNames.ORDER;
+
+import com.google.common.collect.Iterables;
+import com.hedera.mirror.common.domain.entity.EntityId;
+import com.hedera.mirror.rest.model.Links;
+import jakarta.inject.Named;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.domain.Sort.Direction;
+import org.springframework.data.domain.Sort.Order;
+import org.springframework.util.CollectionUtils;
+import org.springframework.web.util.UriComponentsBuilder;
+
+@Named
+@RequiredArgsConstructor
+public class LinkFactoryImpl implements LinkFactory {
+
+    public <T> Links create(List<T> items, Pageable pageable, ParameterExtractor<T> extractor) {
+        var links = new Links();
+        if (items == null || items.isEmpty() || pageable.isUnpaged()) {
```

Comment:
```suggestion
        if (CollectionUtils.isEmpty(items)) {
```

I don't think we should check `isUnpaged` at all, the contract should be if no next link needed, just don't call `create`